Source/scrollrt.cpp

```diff
@@ -1416,7 +1404,7 @@ Displacement GetOffsetForWalking(const AnimationInfo &animationInfo, const Direc
 {
 	// clang-format off
 	//                                           South,        SouthWest,    West,         NorthWest,    North,        NorthEast,     East,         SouthEast,
-	constexpr Displacement StartOffset[8]    = { {   0, -32 }, {  32, -16 }, {  32, -16 }, {   0,   0 }, {   0,   0 }, {  0,    0 },  { -32, -16 }, { -32, -16 } };
+	constexpr Displacement StartOffset[8]    = { {   0, -32 }, {  32, -16 }, {  64,   0 }, {   0,   0 }, {   0,   0 }, {  0,    0 },  { -64,   0 }, { -32, -16 } };
 	constexpr Displacement MovingOffset[8]   = { {   0,  32 }, { -32,  16 }, { -64,   0 }, { -32, -16 }, {   0, -32 }, {  32, -16 },  {  64,   0 }, {  32,  16 } };

```

Comment:
Feels like a really good sign that the offers are not symmetrical.

Seams this function could now be reduced to something like:
```cpp
Displacement GetOffsetForWalking(const AnimationInfo &animationInfo, const Direction dir, bool cameraMode /*= false*/)
{
	//                                         South,     SouthWest,   West,       NorthWest,    North,      NorthEast,   East,      SouthEast,
	constexpr Displacement MovingOffset[8] = { { 0, 32 }, { -32, 16 }, { -64, 0 }, { -32, -16 }, { 0, -32 }, { 32, -16 }, { 64, 0 }, { 32, 16 } };

	float fAnimationProgress = animationInfo.GetAnimationProgress();

	if (cameraMode) {
		fAnimationProgress = -fAnimationProgress;
	} else {
		fAnimationProgress -= 1.F;
	}

	return MovingOffset[static_cast<size_t>(dir)] * fAnimationProgress;
}
```

Reply:
I tested the code and it doesn't work with NorthWest, North and NorthEast.
The problem is that in these directions the logic which tile has the positive or negative in `dPlayer`/`dMonster` is different.
Normally the starting tile is negative and the target tile is positive.
But in NorthWest, North and NorthEast the logic is reversed. 😕 
I think this could be unified, but I also think this is outside of the scope of this pr cause it also affects/alters missile hit logic (another tile would be hit tested).

Reply:
Just wondering, could this difference in behavior be related to how it's easier to kite monsters in certain directions? Like when moving southward, if you get hit stunned, you are moved to the destination tile automatically, but if you are moving northward and get hit stunned, you move back to the source tile?

Reply:
yes, that's another thing to fix after horizontal walk